server/channels/app/plugin_api_test.go

```diff
@@ -2195,3 +2195,123 @@ func TestPluginUploadsAPI(t *testing.T) {
 	require.NotNil(t, manifest)
 	require.True(t, activated)
 }
+
+func TestConfigurationWillBeSavedHook(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	getPluginCode := func(hookCode string) string {
+		return fmt.Sprintf(`
+    package main
```

Comment:
nit: indent here might make it harder to maintain. WI wonder if it would make sense to `go:embed` a file instead?

Reply:
Yes, I admit I thought of that but got a bit lazy :) Thanks for the nit, will have a look.

Reply:
Done, still the improvement is not massive given that the file content is used as template so most of the key logic gets injected at run-time based on the test case.